air/tests/test_module/negative_tests/execution_step.rs

```diff
@@ -0,0 +1,435 @@
+        (par
+            (match 0 1
+                (call "vm_1_peer_id" ("m" "f") ["scalar_1_result"] {var_name})
+            )
+            (ap {var_name} scalar_2)
+        )
+        "#);
+    let mut vm_2 = create_avm(echo_call_service(), vm_2_peer_id);
+    let result = vm_2
+        .runner
+        .call(&script, "", "", vm_2_peer_id, 0, 0, None, <_>::default())
+        .unwrap();
+    let expected_error = CatchableError::VariableNotFound(var_name);
+    assert!(check_error(&result, expected_error));
+}
+
+#[test]
+fn ap_canon_stream_variable_not_found_error() {
+    let vm_2_peer_id = "vm_2_peer_id";
+    let var_name = "#canon".to_string();
+    let script = f!(r#"
+        (par
+            (match 0 1
+                (seq
+                    (call "vm_1_peer_id" ("m" "f") ["scalar_1_result"] $stream)
+                    (canon "vm_1_peer_id" $stream {var_name})
+                )
+            )
+            (ap {var_name} scalar_2)
+        )
+        "#);
+    let mut vm_2 = create_avm(echo_call_service(), vm_2_peer_id);
+    let result = vm_2
+        .runner
+        .call(&script, "", "", vm_2_peer_id, 0, 0, None, <_>::default())
+        .unwrap();
+    let expected_error = CatchableError::VariableNotFound(var_name);
+    assert!(check_error(&result, expected_error));
+}
+
+#[test]
+fn non_array_in_fold_iterable() {
+    let vm_2_peer_id = "vm_2_peer_id";
+    let var_name = "outcome".to_string();
+    let script = f!(r#"
+        (seq
+            (call "{vm_2_peer_id}" ("m" "f") [] {var_name})
+            (fold {var_name} i
+                (seq
+                    (null)
+                    (next i)
+                )
+            )
+        )
+        "#);
+    let unsupported_jvalue = json!({"attr": 42, });
+    let mut vm_2 = create_avm(set_variable_call_service(unsupported_jvalue.clone()), vm_2_peer_id);
+    let result = vm_2.call(&script, "", "", <_>::default()).unwrap();
+    let expected_error = CatchableError::FoldIteratesOverNonArray(unsupported_jvalue, var_name);
+    assert!(check_error(&result, expected_error));
+}
+
+#[test]
+fn non_array_lambda_result_fold_iterable1() {
+    let vm_2_peer_id = "vm_2_peer_id";
+    let var_name = "outcome".to_string();
+    let scalar_int = 42;
+    let script = f!(r#"
+        (seq
+            (call "{vm_2_peer_id}" ("m" "f") [] {var_name})
+            (fold {var_name}.$.int i
+                (seq
+                    (null)
+                    (next i)
+                )
+            )
+        )
+        "#);
+    let unsupported_jvalue = json!({"int": scalar_int, });
+    let mut vm_2 = create_avm(set_variable_call_service(unsupported_jvalue.clone()), vm_2_peer_id);
+    let result = vm_2.call(&script, "", "", <_>::default()).unwrap();
+    let expected_error = CatchableError::FoldIteratesOverNonArray(json!(scalar_int), ".$.int".to_string());
+    assert!(check_error(&result, expected_error));
+}
+
+#[test]
+fn nonstring_in_triplet_for_call() {
```

Comment:
Let's add 6 more test cases where all of these put into other 2 positions in a `call` triplet.

Reply:
Done